pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/OfflineClusterIntegrationTest.java

```diff
@@ -170,9 +170,26 @@ public void setUp()
     // Unpack the Avro files
     List<File> avroFiles = unpackAvroData(_tempDir);
 
-    // Create and upload segments
+    // Create and upload segments. For exhaustive testing, concurrently upload multiple segments with the same name
+    // and validate correctness with parallel push protection enabled.
     ClusterIntegrationTestUtils.buildSegmentsFromAvro(avroFiles, tableConfig, schema, 0, _segmentDir, _tarDir);
-    uploadSegments(getTableName(), _tarDir);
+    // Create a copy of _tarDir to create multiple segments with the same name.
+    File tarDir2 = new File(_tempDir, "tarDir2");
+    FileUtils.copyDirectory(_tarDir, tarDir2);
+
+    List<File> tarDirPaths = new ArrayList<>();
+    tarDirPaths.add(_tarDir);
+    tarDirPaths.add(tarDir2);
+
+    try {
+      uploadSegments(getTableName(), tarDirPaths, TableType.OFFLINE, true);
```

Comment:
Add `Assert.fail()` right after this line since we should expect the exception to be thrown?

Reply:
We might not always hit an exception here because of timing. That's why I did add a fail(). 
However, if we hit an exception, I make sure that the exception is hit because of concurrent uploads.